### ISSUE TITLE
Cannot read property 'getSession' of undefined on Vercel

### ISSUE THREAD

**Issue by CalebDelbridge** [blitz-js/blitz#2772]
### What is the problem?

When I use `getSession` in getServerSideProps on an app deployed on Vercel the app crashes

### Paste all your error logs here:

```
Unhandled error during request: TypeError: Cannot read property 'getSession' of undefined
    at /var/task/node_modules/next/dist/stdlib-server/auth-sessions.js:551:65
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/var/task/node_modules/next/dist/stdlib-server/auth-sessions.js:21:28)
    at _next (/var/task/node_modules/next/dist/stdlib-server/auth-sessions.js:39:17)
    at /var/task/node_modules/next/dist/stdlib-server/auth-sessions.js:44:13
    at new Promise (<anonymous>)
    at /var/task/node_modules/next/dist/stdlib-server/auth-sessions.js:36:16
    at _getSessionKernel (/var/task/node_modules/next/dist/stdlib-server/auth-sessions.js:648:30)
    at getSessionKernel (/var/task/node_modules/next/dist/stdlib-server/auth-sessions.js:651:30)
    at /var/task/node_modules/next/dist/stdlib-server/auth-sessions.js:250:35
```

### Paste all relevant code snippets here:

```
export const getServerSideProps: GetServerSideProps = async ({ req, res }) => {
  const session = await getSession(req, res)
  if (!session.userId) {
    return {
      redirect: {
        destination: Routes.LoginPage().pathname,
        permanent: false,
      },
    }
  } else {
    const calendar = await invokeWithMiddleware(getLatestCalendar, {}, { req, res })
    return {
      redirect: {
        destination: formatRoute(Routes.CalendarPage({ calendarId: calendar.id })),
        permanent: false,
      },
    }
  }
}
```

### What are detailed steps to reproduce this?

1. Create a new blitz app
2. Create a page using the getServerSideProps snippet above
3. Host the app on vercel
4. Navigate to route

### Run `blitz -v` and paste the output here:

```
macOS Big Sur | darwin-x64 | Node: v14.15.4

blitz: 0.40.0-canary.9 (local)

  Package manager: yarn 
  System:
    OS: macOS 11.6
    CPU: (4) x64 Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz
    Memory: 45.71 MB / 4.00 GB
    Shell: 3.1.2 - /usr/local/bin/fish
  Binaries:
    Node: 14.15.4 - /private/var/folders/5m/n35c8kx14n3c7sxzgk2h5jkm0000gn/T/xfs-6a015d00/node
    Yarn: 3.0.2 - /private/var/folders/5m/n35c8kx14n3c7sxzgk2h5jkm0000gn/T/xfs-6a015d00/yarn
    npm: 7.0.15 - /usr/local/bin/npm
    Watchman: Not Found
  npmPackages:
    @prisma/client: latest => 3.0.2 
    blitz: 0.40.0-canary.9 => 0.40.0-canary.9 
    prisma: 3.0.1 => 3.0.1 
    react: alpha => 18.0.0-alpha-b6ff9ad16-20210820 
    react-dom: alpha => 18.0.0-alpha-b6ff9ad16-20210820 
    typescript: ~4.3 => 4.3.5
```

**Comment by adavijit** [comment]
@CalebDelbridge , I would like to take this issue

**Comment by CalebDelbridge** [comment]
Great, thanks @adavijit!

**Comment by adavijit** [comment]
@CalebDelbridge , Could you please assign to me?

**Comment by CalebDelbridge** [comment]
I think you would have to get a maintainer or someone with that ability to do that, but as long as you comment that you claim this issue, you should be good and that no one will try to fix it before you.

**Comment by flybayer** [comment]
@adavijit ah, I had just assigned it to me before your comments loaded.

Have you made any progress on it? It's actually very tricky and is caused by Vercel's magic file tracing. I've dealt with it before, so probably far easier for me to fix.

**Comment by flybayer** [comment]
@adavijit I've fixed in #2787. I want to get a new stable release published today, so I went ahead and fixed.
